### PR TITLE
Add a method to fetch test issues from FTL

### PR DESCRIPTION
**Pull request by rnakade**
Add a method to fetch test issues for a testMatrix from FTL.

### CODE REVIEW

File: AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt [hunk 1]
```diff
@@ -280,6 +281,19 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    override suspend fun getTestMatrixTestIssues(testMatrix: TestMatrix): List<String> {
+        val steps = testExecutionStore.getTestExecutionSteps(testMatrix)
+        val testIssues = steps.flatMap {
```
Comment (liutikas) [inline AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt:286]: We should also collect issue type

https://firebase.google.com/docs/test-lab/reference/toolresults/rest/v1beta3/projects.histories.executions.steps#Type

and severity
https://firebase.google.com/docs/test-lab/reference/toolresults/rest/v1beta3/projects.histories.executions.steps#Severity

so we could know how to report it back to AnTS

Comment (liutikas) [reply AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt:286]: we should ideally also know which step it happened in, so we can attach the issue to the correct work unit.

Comment (rnakade) [reply AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt:286]: The errorTypes accepted by AnTS are vastly different from what FTL provides. Most of the issues coming from FTL will fall under the "INFRA_ERROR" type. That's why I wasn't sure how useful it'd be to report back. But, I can certainly add these fields to the response.

As for the step, we don't create separate workunits for different shards. So, it going to be very difficult to use the step info to associate failures with that level of granularity.

Comment (liutikas) [reply AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt:286]: ACK on shards, we can skip step bits.

severity and type still seem useful. We'll likely only want to warn on severe and warning

Comment (rnakade) [reply AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt:286]: Updated. Please review again. Thanks!

File: AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerService.kt [hunk 2]
```diff
@@ -345,4 +353,22 @@ interface TestRunnerService {
...
+     */
+    data class TestIssue(
+        /**
+         * Error message describing the issue
+         */
+        val errorMessage: String,
+        /**
+         * Severity of issue
+         */
+        val severity: String?,
```
Comment (liutikas) [inline AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerService.kt:368]: Is there a reason we are using strings and not types that are generated in dev.androidx.ci.generated.ftl.* ?

Comment (rnakade) [reply AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerService.kt:368]: Yeah, those classes are internal and can't be used here directly. That's why we always create data classes here to only expose the relevant fields and not the entire output.

**Review by liutikas — APPROVED** [review]

**Pull request merged**